internal/mq/msgstream/mqwrapper/pulsar/pulsar_consumer.go

```diff
@@ -52,20 +49,6 @@ func (pc *Consumer) Chan() <-chan mqwrapper.Message {
 	if pc.msgChannel == nil {
 		pc.once.Do(func() {
 			pc.msgChannel = make(chan mqwrapper.Message, 256)
-			// this part handles msgstream expectation when the consumer is not seeked
```

Comment:
how we can move this behaviour?

Reply:
if write 1 entity and your checkpoint is not generated yet.
Then without seek to earliest you will miss one entity though?

Reply:
Yep,  we will add a initialize offset when creating each virtual channel, I have confirmed with @congqixia